asf_heat_pump_suitability/research/exploratory/sample_buildings_for_labelling.py

```diff
@@ -0,0 +1,401 @@
+from asf_heat_pump_suitability.getters import load_boundaries, load_tree_input
+from asf_heat_pump_suitability.pipeline.transform import uprns
+
+from datetime import date
+
+today = date.today().strftime("%Y%m%d")
+
+# %%
+seed = 10
+
+# %%
+labellers = ["reindeer", "springbok", "anteater", "emu"]
+
+# %%
+# Load Plymouth buildings and residential UPRNs with property type label
+buildings_gdf = gpd.read_file(
+    "s3://asf-heat-pump-suitability/local_heat_planning/inputs/v042025_OSOpenMapLocal_geometries_selected/SX/SX_Building.shp"
+)
+uprns_df = pl.read_parquet(
+    f"s3://asf-heat-pump-suitability/local_heat_planning/outputs/plymouth_residential_uprns_with_flats.parquet"
+)
+
+# Join buildings to UPRNs
+uprns_gdf = (
+    uprns.generate_gdf_uprn_coords(df=uprns_df)
+    .sjoin(buildings_gdf, how="inner", predicate="within")
+    .drop(columns=["index_right", "FEATCODE"])
+)
+uprns_df = pl.from_pandas(uprns_gdf.drop(columns="geometry")).rename(
+    {"ID": "building_id"}
+)
+
+# %% [markdown]
+# ## Original random samples of buildings containing flats from Plymouth
+
+# %%
+# Sample 1
+n = 200
+pl.set_random_seed(seed)
+
+# Identify all buildings containing any flats
+buildings_containing_flats = uprns_df.filter(pl.col("property_type_flat"))[
+    "building_id"
+].unique(maintain_order=True)
+
+# Filter to buildings with flats, then get total count of flats and UPRNs per building
+sample_df = (
+    uprns_df.filter(
+        pl.col("building_id").is_in(buildings_containing_flats),
+    )
+    .group_by("building_id")
+    .agg(
+        pl.col("property_type_flat").sum().alias("n_flats"),
+        pl.col("UPRN").count().alias("n_total"),
+        # Take a random sample from any building with >4 flats
+    )
+    .filter(pl.col("n_flats") > 4)
+    .sort(by="building_id")
+    .sample(n=n, seed=seed)
+)
+
+# Merge sample buildings with their geometries
+sample_gdf = buildings_gdf[["ID", "geometry"]].merge(
+    sample_df.to_pandas(), how="inner", left_on="ID", right_on="building_id"
+)
+print(f"Sample length: {len(sample_gdf)}")
+
+# Convert to 4326 projection for plotting on MyMaps
+sample_4326_gdf = sample_gdf.to_crs(epsg=4326)
+# Get centroid coordinates of each building to geolocate on google maps
+sample_4326_gdf = sample_4326_gdf.merge(
+    sample_4326_gdf.centroid.get_coordinates(),
+    how="left",
+    left_index=True,
+    right_index=True,
+)
+# Create google maps link
+sample_4326_gdf["url"] = sample_4326_gdf.apply(
+    lambda row: f"https://www.google.com/maps/search/?api=1&query={row['y']},{row['x']}",
+    axis=1,
+)
+
+# Save as kml file
+kml = simplekml.Kml()
+for idx, r in sample_4326_gdf.iterrows():
+    pol = kml.newpolygon(
+        name="unlabelled",
+        description=f"Location: https://www.google.com/maps/search/?api=1&query={r['y']},{r['x']}\nN flats: {r['n_flats']}\nN total: {r['n_total']}",
+        outerboundaryis=list(r["geometry"].exterior.coords),
+    )
+    pol.style.polystyle.color = "9939FF14"
```

Comment:
this is not an hex, but I guess it's something similar to set the color of your pologyon?